File: solutions/node/src/shared/Ids.ts
import { randomUUID } from 'node:crypto';

export type IdGenerator = () => string;

export const newId: IdGenerator = () => randomUUID();

// Deterministic ids for tests and the e2e script: sequence('WO-', 2208) -> WO-2208, WO-2209 ...
export function sequence(prefix: string, start: number): IdGenerator {
  let next = start;
  return () => `${prefix}${next++}`;
}

File: starter/python/src/shared/ids.py
import uuid
from typing import Callable

IdGenerator = Callable[[], str]


def new_id() -> str:
    return str(uuid.uuid4())


def sequence(prefix: str, start: int) -> IdGenerator:
    """Deterministic ids for tests and the e2e script: sequence('WO-', 2208)() -> 'WO-2208', 'WO-2209' ..."""
    counter = iter(range(start, 10**9))
    return lambda: f"{prefix}{next(counter)}"
